src/yggdrasil/api.go

```diff
@@ -363,6 +363,22 @@ func (c *Core) SetNodeInfo(nodeinfo interface{}, nodeinfoprivacy bool) {
 	c.router.nodeinfo.setNodeInfo(nodeinfo, nodeinfoprivacy)
 }
 
+// GetMaximumSessionMTU returns the maximum allowed session MTU size.
+func (c *Core) GetMaximumSessionMTU(mtu uint16) uint16 {
+	return c.router.sessions.myMaximumMTU
+}
+
+// SetMaximumSessionMTU sets the maximum allowed session MTU size. The return
+// value contains the actual set value, since Yggdrasil will not accept MTUs
+// below 1280 bytes. The default value is 65535 bytes.
+func (c *Core) SetMaximumSessionMTU(mtu uint16) uint16 {
+	if mtu < 1280 {
+		mtu = 1280
+	}
+	c.router.sessions.myMaximumMTU = mtu
```

Comment:
Like above, this needs to go through a `phony.Block` through the router to make it thread safe. Also, I'm not sure we should be enforcing the 1280 limit here, since that applies to IPv6 only. Maybe the tuntap should enforce that limit, and we just trust it here (and enforce the 65535 limit, but that's implicit from the `uint16` already).

I think the router also needs to send a message to each active session to have them update their local `sinfo.myMTU` value.